flambe/vision/classification/task.py

```diff
@@ -0,0 +1,124 @@
+from typing import Optional
+
+import torch.nn as nn
+
+from flambe.dataset import TabularDataset
+from flambe.nn import Encoder
+from flambe.metric import Accuracy
+from flambe.field import LabelField
+from flambe.task import Training
+from flambe.optim.optimizer import Optimizer
+from flambe.optim.lr_scheduler import LRScheduler
+from flambe.nlp.classification import TextClassification
+
+
+class ImageClassifier(Encoder):
+    """Implements a simple image classifier."""
+
+    def __init__(self, encoder: Encoder, output_size: int, dropout: float = 0) -> None:
+        """A simple image classifier."""
+        super().__init__()
+        self.output_size = output_size
+        self.encoder = encoder
+        self.drop = nn.Dropout(dropout)
+        self.output_layer = nn.Linear(encoder.output_dim, output_size)
+
+    @property
+    def input_dim(self) -> int:
+        """Get the size of the last dimension of an input."""
+        return self.encoder.input_dim
+
+    @property
+    def output_dim(self) -> int:
+        """Get the size of the last dimension of an output."""
+        return self.output_size
+
+    def forward(self, data):  # type: ignore
+        """Run a forward pass from shape (B x H) to (B x O)."""
+        return self.output_layer(self.drop(self.encoder(data).flatten(1)))
+
+
+class ImageClassification(TextClassification):
```

Comment:
Why inherit from `TextClassification` ? I think this is a typo.

Reply:
it saves some code, but I could build a Classification task instead which they both inherit from